swarm_nl/src/core/mod.rs

```diff
@@ -720,6 +741,20 @@ impl<T: EventHandler + Clone + Send + Sync + 'static> Core<T> {
 		self.keypair.public().to_peer_id()
 	}
 
+	/// Return an iterator to the buffered network layer events and consumes them
+	pub async fn events(&mut self) -> IntoIter<NetworkEvent> {
+		let events = self.event_queue.into_inner().await.into_iter();
+
+		// Clear all buffered events
+		self.event_queue.drain().await;
+		events
```

Comment:
When you call the `events()` function on the network core, it returns an iterator for the current events in the queue and then clears the queue since its events have been accessed by the caller. If you prefer not to consume all the events at once, you can use the `next_event()` function, which returns only the event at the front of the queue.